op-node/rollup/driver/state.go

```diff
@@ -257,39 +249,46 @@ func (s *Driver) eventLoop() {
 	// L1 chain that we need to handle.
 	reqStep()
 
+	blockTime := time.Duration(s.config.BlockTime) * time.Second
+
+	var sequenceErr error
+	var sequenceErrTime time.Time
+	sequencerTimer := time.NewTimer(0)
+	var sequencerCh <-chan time.Time
+
 	for {
-		select {
-		case <-l2BlockCreationTickerCh:
-			s.log.Trace("L2 Creation Ticker")
-			s.snapshot("L2 Creation Ticker")
-			reqL2BlockCreation()
-
-		case <-l2BlockCreationReqCh:
-			s.snapshot("L2 Block Creation Request")
-			if !s.idleDerivation {
-				s.log.Warn("not creating block, node is deriving new l2 data", "head_l1", s.l1State.L1Head())
-				break
-			}
-			ctx, cancel := context.WithTimeout(ctx, 20*time.Minute)
-			err := s.createNewL2Block(ctx)
-			cancel()
-			if err != nil {
-				s.log.Error("Error creating new L2 block", "err", err)
-				s.metrics.RecordSequencingError()
-				break // if we fail, we wait for the next block creation trigger.
+		var sealNext bool
+		if s.driverConfig.SequencerEnabled {
+			delay, seal := s.sequencer.PlanNextSequencerAction(sequenceErr)
+			if sequenceErr != nil && time.Since(sequenceErrTime) > delay {
+				sequenceErr = nil
 			}
+			sequencerCh = sequencerTimer.C
+			sequencerTimer.Reset(delay)
```

Comment:
```
			if x := now.Add(delay); x.Sub(sequencePlanTime).Abs() > time.Millisecond {
				sequencerTimer.Reset(delay)
				sequencePlanTime = x
			}
```
We can do something like this to avoid resets on insignificant changes, but isn't it a bit premature optimization?

It's really useful to check it every loop, since we want to change the sequencing timing based on forkchoice changes. The function looks like only a few stack allocations and if statements, and one call to time.Now(), not really that heavy. And the `Reset` benchmarks at 50 ns/op, with 0 allocations.

Reply:
```
func BenchmarkTimer(b *testing.B) {
	timer := time.NewTimer(time.Second)
	b.ReportAllocs()
	b.ResetTimer()
	for i := 0; i < b.N; i++ {
		timer.Reset(time.Second)
		timer.Reset(time.Minute)
	}
}
```

Reply:
Fixed the timer issue by draining it, and changed the code to re-schedule only when the head changed since last planned action.